crossfit/backend/torch/hf/model.py

```diff
@@ -13,30 +13,57 @@
 # limitations under the License.
 
 import gc
-import os
 from functools import lru_cache
 
-import joblib
 import numpy as np
 import torch
-from sklearn.linear_model import LinearRegression
-from tqdm import tqdm
 from transformers import AutoConfig, AutoModel, AutoTokenizer
 
+from crossfit.backend.torch.hf.memory_curve_utils import fit_memory_estimate_curve
 from crossfit.backend.torch.model import Model
-from crossfit.dataset.home import CF_HOME
-from crossfit.utils.model_adapter import adapt_model_input
 
 
 class HFModel(Model):
-    def __init__(self, path_or_name: str, max_mem_gb: int = 16, training=False):
+    def __init__(
+        self,
+        path_or_name: str,
+        max_mem_gb: int = 16,
+        training: bool = False,
+        start_batch_size: int = 1,
+        end_batch_size: int = 2048,
+        batch_size_increment: int = 256,
+        start_seq_len: int = 1,
+        seq_len_increment: int = 64,
+    ):
         super().__init__(path_or_name, max_mem_gb)
+        self.start_batch_size = start_batch_size
+        self.end_batch_size = end_batch_size
+        self.batch_size_increment = batch_size_increment
+        self.start_seq_len = start_seq_len
+        self.seq_len_increment = seq_len_increment
 
+        model = self.load_model("cuda") if not training else None
```

Comment:
One more thing I thought of actually. Loading the model can take a noticeable amount of time (like 20-30 seconds for Llama 7B on my machine). There is this line of code that seems to check to see if the memory info already exists before running the tests:
```python
    if os.path.exists(mem_model_path):
        return joblib.load(mem_model_path)
```

What do you think about moving the `load_model` call after this early return statement? It should save a bit of time. Otherwise, I've noticed that the model gets loaded for the memory test that doesn't happen, and then it just gets reloaded again on all the workers for the inference which wastes a bit of time.

Reply:
Done.